File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.toLowerCase();
    let filteredString = "";
    let reverseString = "";
    for(let i=0;i<s.length;i++){
        if((s[i].charCodeAt()>="a".charCodeAt() && s[i].charCodeAt()<="z".charCodeAt()) || (s[i].charCodeAt()>="0".charCodeAt() && s[i].charCodeAt()<="9".charCodeAt())){
            filteredString+=s[i];
            reverseString = s[i]+reverseString

        }
    }
    console.log(filteredString);
    return filteredString  === reverseString;
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s=s.toLowerCase();
        int start =0;
        int end = s.length()-1;
        while(start<end){
            char ch1 = s.charAt(start);
            char ch2 = s.charAt(end);
            if(!Character.isLetterOrDigit(ch1)){
                start++;

            }else if(!Character.isLetterOrDigit(ch2)){
                end--;
            }else{ 
                if(ch1!=ch2){
                    return false;
                }
                start++;
                end--;

            }


        }
        return true;
    }
}